### PR TITLE
SNOW-870225: remove vendored arrow

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-870225

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/connector/cursor.py [hunk 1]
```diff
@@ -1524,3 +1506,31 @@ def __init__(self, connection) -> None:
             connection,
             use_dict_result=True,
         )
+
+
+def __getattr__(name):
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/cursor.py:1511]: I don't see tests for this. Please at least manually test that this works as expected

**Review by sfc-gh-mkeller — APPROVED** [review]
Wow, feels like the end of an era for `snwoflake-connector-python`. Big thanks for this @sfc-gh-aling!
This is going to be a great release!

:ship: :ship:

**Comment by sfc-gh-mkeller** [comment]
@xhochy just wanted to make sure you are aware of what's coming, the release of this is planned for the end of next week

File: src/snowflake/connector/cursor.py (hunk 1 above)
Comment (sfc-gh-aling) [reply src/snowflake/connector/cursor.py:1511]: I have added test for this, it's working as expected

**Pull request merged**